frontend/src/components/game/Editor.tsx

```diff
@@ -6,10 +6,11 @@ import Language, { fromString, languageToEditorLanguage } from '../../api/Langua
 import { DefaultCodeType } from '../../api/Problem';
 
 type EditorProps = {
```

Comment:
There are a number of small edits on this page - what these are fixing / changing exactly?

Reply:
These allow the editor to distinguish between editing and previewing mode - so if users don't pass in an onChange or onLanguageChange handler, it will set the code and language to be view only. And then also the defaultCode is for the previewing code which I think was talked about in another comment. 

As mentioned this solution works although is a tad bit hacky lol, so I'm open to any suggestions if you have ideas for streamlining it

Reply:
Ah, so is this what's used for viewing the code on the results page then? If so, that sounds good, I think this is ok